# Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [1]:
import numpy as np
import pandas as pd

In [5]:
dataset = pd.DataFrame({
    'study hours per day' : np.array([1,2,3,2,0,4,5]),
    'score (Out of 10)' : np.array([2,4,7,3,1,8,9])
})

In [6]:
dataset

study hours per day  score (Out of 10)
0                    1                  2
1                    2                  4
2                    3                  7
3                    2                  3
4                    0                  1
5                    4                  8
6                    5                  9

In [9]:
dataset.corr(method='pearson')

study hours per day  score (Out of 10)
study hours per day             1.000000           0.973337
score (Out of 10)               0.973337           1.000000

Correlation value between number of study hours per day and the score of students in exam is 0.973337 which is very close to 1.

This indicates that this to variables have nearly perfect positive of correlation. 

They have a strong positive linear correlation.
 
As number of study hours per day increases, score of students in exam increases and vice versa.


# Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [15]:
dataset = pd.DataFrame({
    'Amount of sleep(Hours)' : [6,7,8,5,8,7,6],
    'Job Satisfaction' : [8,9,8,6,9,8,7]
})

In [16]:
dataset.corr(method='spearman')

Amount of sleep(Hours)  Job Satisfaction
Amount of sleep(Hours)                1.000000          0.750139
Job Satisfaction                      0.750139          1.000000

Correlation value of two variable is 0.750139 which is very near to 1.
This indiacates that there is a strong positive correlation between two variables. 
Relationship is monotonic.

# Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [56]:
np.random.seed(101)
dataset = pd.DataFrame({
    'hours' : np.random.randint(1, 10, 50),
    'bmi' : np.random.uniform(18, 30, 50)
})

In [55]:
dataset.head()

hours        bmi
0      2  23.285390
1      7  25.406736
2      8  24.536899
3      9  22.997807
4      5  26.755837

In [54]:
dataset.corr(method='spearman')

hours       bmi
hours  1.000000  0.057489
bmi    0.057489  1.000000

In [53]:
dataset.corr(method='pearson')

hours       bmi
hours  1.000000  0.054015
bmi    0.054015  1.000000

Both spearman and pearson correlation are giving nearly same result. Both are nearly zero which indiacates there is no relation between this two variables. This result is because i created a random dataset which is not accurate. 

# Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [58]:
np.random.seed(100)
dataset = pd.DataFrame({
    'Television' :np.random.randint(1,5,50),
    'physical Activity' : np.random.randint(1,5,50)
})

In [61]:
dataset.corr('pearson')

Television  physical Activity
Television            1.00000           -0.40714
physical Activity    -0.40714            1.00000

Variables are negatively correlated.

# Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:

In [62]:
dataset = pd.DataFrame({
    'Age(Years)' : [25,42,37,19,31,28],
    'Soft drink Preference' : ['Coke', 'Pepsi', 'Mountain dew', 'Coke', 'Pepsi', 'Coke']
})

In [71]:
# encoding the data using nominal encoding
from sklearn.preprocessing import LabelEncoder
encoder = LabelEncoder()
data = encoder.fit_transform(dataset['Soft drink Preference'])
dataset['Soft drink Preference encoded'] = data

In [74]:
dataset.head()

Age(Years) Soft drink Preference  Soft drink Preference encoded
0          25                  Coke                              0
1          42                 Pepsi                              2
2          37          Mountain dew                              1
3          19                  Coke                              0
4          31                 Pepsi                              2

In [72]:
dataset.corr('spearman')

C:\Users\hp\AppData\Local\Temp\ipykernel_20268\132823682.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  dataset.corr('spearman')


Age(Years)  Soft drink Preference encoded
Age(Years)                       1.000000                       0.833238
Soft drink Preference encoded    0.833238                       1.000000

# Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [92]:
dataset = pd.DataFrame({
    'sales calls per day' : np.random.randint(1,10,30),
    'sales per week' : np.random.randint(10,30,30)
})

In [93]:
dataset.corr(method='spearman')

sales calls per day  sales per week
sales calls per day             1.000000        0.298148
sales per week                  0.298148        1.000000